diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileStructureRule.kt

```diff
@@ -0,0 +1,161 @@
+
+        // checking order
+        listOfNotNull(copyrightComment, headerKdoc, fileAnnotations).forEach { astNode ->
+            val (afterThisNode, beforeThisNode) = astNode.getSiblingBlocks(copyrightComment, headerKdoc, fileAnnotations, packageDirectiveNode)
+            val isPositionIncorrect = (afterThisNode != null && !node.isChildAfterAnother(astNode, afterThisNode))
+                    || !node.isChildBeforeAnother(astNode, beforeThisNode)
+
+            if (isPositionIncorrect) {
+                FILE_INCORRECT_BLOCKS_ORDER.warnAndFix(configRules, emitWarn, isFixMode, astNode.text.lines().first(), astNode.startOffset) {
+                    node.moveChildBefore(astNode, beforeThisNode, 2)
+                }
+            }
+        }
+
+        // checking empty lines
+        arrayOf(copyrightComment, headerKdoc, fileAnnotations, packageDirectiveNode, importsList).forEach { astNode ->
+            astNode?.treeNext?.apply {
+                if (elementType == WHITE_SPACE && text.count { it == '\n' } != 2) {
+                    FILE_NO_BLANK_LINE_BETWEEN_BLOCKS.warnAndFix(configRules, emitWarn, isFixMode, astNode.text.lines().first(), astNode.startOffset) {
+                        (this as LeafPsiElement).replaceWithText("\n\n${text.replace("\n", "")}")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun checkImportsOrder(node: ASTNode) {
+        val imports = node.getChildren(TokenSet.create(IMPORT_DIRECTIVE)).toList()
+
+        // importPath can be null if import name cannot be parsed, which should be a very rare case, therefore !! should be safe here
+        imports.filter { (it.psi as KtImportDirective).importPath!!.isAllUnder }.forEach {
+            FILE_COLLAPSED_IMPORTS.warn(configRules, emitWarn, isFixMode, it.text, it.startOffset)
+        }
+
+        val sortedImports = imports.sortedBy { it.text }
+        if (sortedImports != imports) {
+            FILE_UNORDERED_IMPORTS.warnAndFix(configRules, emitWarn, isFixMode, fileName, node.startOffset) {
+                // move all commented lines among import before imports block
+                node.getChildren(null).filterIndexed { index, astNode ->
+                    index < node.getChildren(null).indexOf(imports.last()) && astNode.elementType == EOL_COMMENT
+                }.forEach {
+                    node.treeParent.addChild(it.clone() as ASTNode, node)
+                    node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node)
+                }
+
+                node.removeRange(imports.first(), imports.last())
+                sortedImports.forEachIndexed { index, importNode ->
+                    if (index != 0) {
+                        node.addChild(PsiWhiteSpaceImpl("\n"), null)
+                    }
+                    node.addChild(importNode, null)
+                }
+            }
+        }
+    }
+
+    private fun ASTNode.getSiblingBlocks(copyrightComment: ASTNode?,
```

Comment:
I guess we can increase coverage if you would write tests for such functions like this

Reply:
Specifically this method, imo, doesn't really need testing, it acts as a map with dynamic values. The next method, `ASTNode.moveChildBefore`, should be tested, but I actually have already implemented the same method with tests in #6. We should merge that PR first and I'll reuse that method here.